src/NuGet.Core/NuGet.Commands/RestoreCommand/DependencyGraphResolver.cs

```diff
@@ -913,6 +929,12 @@ private async Task<Dictionary<LibraryDependencyIndex, ResolvedDependencyGraphIte
             // Stores any evictions to process
             Dictionary<LibraryRangeIndex, (LibraryRangeIndex[], LibraryDependencyIndex, LibraryDependencyTarget)> evictions = new Dictionary<LibraryRangeIndex, (LibraryRangeIndex[], LibraryDependencyIndex, LibraryDependencyTarget)>(EvictionsDictionarySize);
 
+            // Keeps track of the number of times the entire walk is started over
+            int? restartCount = NuGetEventSource.IsEnabled ? -1 : null;
+
+            // Keeps track of the total number of items that have been queued for processing
+            int? totalQueuedItemCount = NuGetEventSource.IsEnabled ? 0 : null;
```

Comment:
I don't understand the intent of making these counters nullable.

Two ints would normally increase the stack size by 8 bytes. Making them nullable makes that even larger, although I don't know the runtime implementation details well enough to know by exactly how much (but each nullable needs a bool, so maybe 2 more bytes).

Plus each time the counter is incremented, the compiler needs to emit "if not null" guards. Since CPUs have branch predictors, if the CPU guesses wrong, the pipeline needs to be flushed.  All this to say that having nullable counters probably has worse theoretical perf than having a counter that's only written to, never read.

Reply:
If we don't think it matters one way or the other, which implementation should we go with?

Reply:
I have removed the `int?`